update your-first-algo.md (#381)
algorithmiaio/dev-center: _pages/algorithm-development/your-first-algo.md

To run this algorithm, first hit the "Build" button on the top-right corner of the algorithm editor. At the bottom of the page in the console you'll see a confirmation that it has been built (compiled) successfully, along with the commit hash. Until you've published your algorithm, the version number will be a hash such as `4be0e18fba270e4aaa7cff20555268903f69a11b`, and only you will be able to call this version. After you've published an algorithm, it will be given a `major.minor.revision` number as described in the [Versioning Documentation]({{site.baseurl}}/basics/versioning).

Compiling your algorithm will also save your work. Note that the first time you compile your algorithm it might take some time, while subsequent compiles will be quicker.
{: .notice-info}

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/algo_dev_lang/algorithm_console_python.png" alt="Algorithm console Python" class="screenshot">

Once you have finished editing and want to run the algorithm, click the green **Build** button at the top-right corner of the editor. This will save your algorithm by committing your code to your personal git repository and will try to compile your code.

> Once your algorithm has successfully compilied, you can test it out by passing input though the console at the bottom of the screen. For instance, after compiling this algorithm, you can test the "Hello World" code by typing in "World" in the console:

Reviewer: ```suggestion
Once your algorithm has successfully compiled, you can test it out by passing input through the console at the bottom of the screen. For instance, after compiling this algorithm, you can test the "Hello World" code by typing in "World" in the console:
```

To learn about the various data sources that Algorithmia supports natively, visit <a href="{{site.baseurl}}/data">Connecting Your Data</a> or check out our reference guide on the available methods using <a href="http://docs.algorithmia.com/#data-api-specification">Algorithmia's Data API</a>.

If you are authoring an algorithm and using the Data API, avoid using the `.my` pseudonym in the source code. When the algorithm is executed, `.my` will be interpreted as the user name of the user who called the algorithm, rather than the author’s user name.
{: .notice-warning}

## Calling Your Algorithm
To call your algorithm during development and testing, you can use the commit hash to call it locally using the language client of your choice. If you are developing within the Algorithmia Web IDE, you can use the built-in console for testing. 

To call a private version of an algorithm you own, you must use a fully specified semantic version or a version hash or you will get an error. For more information see the details in the <a href="https://algorithmia.com/developers/api/#api-specification">API Docs</a>
{: .notice-warning}

## Publishing Your Algorithm

> The last step is publishing your algorithm. The best part of hosting your code on Algorithmia is that users can access it via an API that takes only a few lines of code to use! Here is what you can set when publishing your algorithm:

Reviewer: Lets make this a little more enterprise friendly - something like "The last step is publishing your algorithm. Deploying your code to your Algorithmia cluster will provide you with an endpoint to share that will scale automatically no matter how often the API is called" - this isn't great, but something along those lines

Author: how about: "The last step is publishing your algorithm. Once your code is deployed on your Algorithmia cluster, you will have a shareable endpoint that will scale automatically to handle request load, no matter how often the API is called."

Reviewer: perfect use that!

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/algo_dev_lang/publish_algorithm.png" alt="Publish an algorithm" class="screenshot img-sm">

There are 3 steps to publishing an algorithm: documenting changes, adding example input and output, and configuring algorithm settings.

The **Changes** step shows you your commit history and allows you to add release notes.

The **Example Input/Output** step is where you'll enter your sample input, which will be displayed in the Overview tab on the algorithm home page, where users may test your algorithm. This is a good time to test your algorithm with all of the types of inputs that your algorithm accepts, to verify that the output is as expected, because users will be able to test your algorithm with their own inputs.

During the **Configure Settings** step you can select whether your algorithm will be publicly callable or for private use only. Under "Semantic Version" you can choose which kind of release characterizes your changes: Major, Minor, or Revision. Finally, you can enable the collection of metrics using Algorithmia Insights by checking the "Insights" box.

> If you are satisfied with your algorithm code, behavior, and settings, go ahead and hit "Publish". Congratulations, you’re an algorithm developer!

Reviewer: Again making it a bit more enterprise focused, here let's change "Congratulations, you’re an algorithm developer!" to something like "Congratulations, you just deployed your machine learning model/code into production and it's immediately ready for use!" - or something along those lines

Author: yeah I like that, something short: "Congratulations, you just deployed your code into production and it can be called immediately!"

Reviewer: cool, feel free to change that and once all these changes are done, I'll approve the PR

### Editing Your Algorithm

Your published algorithm can be edited from the browser, where you can modify the source code, save your work, compile, and publish the algorithm to be available through the API. You can also <a href="{{site.baseurl}}/algorithm-development/source-code-management#editing-your-algorithm-source-locally">use Git to push directly to Algorithmia</a> from your current workflow.

### Calling Other Algorithms

When an algorithm is called from within another algorithm, the algorithm called internally is considered to have been made from within the same session as the first algorithm call. Because of this, you won't need to set an API key within your algorithm code. For example, instead of setting the API key on the client like you would when consuming an algorithm, you can call an algorithm internally and pass the input like so:

{% highlight python %}
if 'url' in input:
  text = Algorithmia.algo('util/Html2Text/0.1.3').pipe(input['url']).result
{% endhighlight %}

> This example shows an algorithm that checks the type of input passed in, and if it is a URL, calls the Html2Text algorithm to pull down code from the webpage. Compute associated with both the original algorithm call and the internal algorithm call is tracked and attributed to the original caller's user account.

Reviewer: ```suggestion
This example shows an algorithm that checks the type of input passed in, and if it is a URL, calls the Html2Text algorithm to pull down code from the webpage. Compute associated with both the original algorithm call and the internal algorithm call is tracked and attributed to the original caller's account.
```

Reviewer: Just because we don't want to confuse "user account" for organization owned assets

Author: I accidentally resolved before committing. So I think this still needs to be committed as it's now "outdated".

## (Optional) Deleting your Algorithm

If your algorithm has never been published, or has only been published for private use, you have the option to delete it when it is no longer needed. To do so, go to your algorithm's page, click on the **Settings** tab, and find the red **Delete** button at the bottom.

<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/algo_dev_lang/delete_algorithm.png" alt="Delete algorithm" class="screenshot img-sm">

If you have ever published your Algorithm with "Public" selected, it cannot be deleted. This is to ensure that others' code will not break if they have written algorithms or client code that depends on your algorithm.

## Wrapping Up

> Now that you've published your first simple algorithm on the platform, you are empowered to develop more complex and useful algorithms. You can write them from scratch, adopt and modify open-source algorithms, or even chain multiple pre-existing algorithms together to make a micro service that you can call with the Algorithmia API! Get inspired by checking out our <a href="{{site.baseurl}}/tutorials/recipes">Recipes</a>.

Reviewer: ```suggestion
Now that you've published your first simple algorithm on the platform, you are empowered to develop more complex and useful algorithms. You can write them from scratch, adopt and modify open-source algorithms, or even chain multiple pre-existing algorithms together to make a micro service that you can call with the Algorithmia API! 
```

Reviewer: Recipes section is slated for removal - if time allows I'd like the last section mention something like "chain together algorithms to take advantage of our microservice architecture" or something like that - maybe even mentioning use case like data pre-processing algo combined with image processing or something like that to showcase pipelining

Author: Yeah that sounds like a great use case / demo to build. I have added it to my personal TODO board, so I think we can consider it resolved here.
